### DISCUSSION TITLE
CLI not working in automator or applescript

### DISCUSSION THREAD

**Discussion by theboyler** [samschott/maestral#524] [Q&A] [answered]
I'm trying to develop an automator service on Mac so I can refight-click on a file and generate a sharelink for that file. I've got the path to the file, I can truncate it in AppleScript and add the current date + one week, and then try and run it through a 'do shell script' command, and I get an error from Maestral:

> Traceback (most recent call last):
>   File "Contents/Resources/app/maestral_cocoa/__main__.py", line 41, in <module>
>   File "Contents/Resources/app/maestral_cocoa/__main__.py", line 36, in freeze_support_cli
>   File "Contents/Resources/app_packages/click/core.py", line 1128, in __call__
>   File "Contents/Resources/app_packages/click/core.py", line 1034, in main
>   File "Contents/Resources/app_packages/click/_unicodefun.py", line 100, in _verify_python_env
> RuntimeError: Click will abort further execution because Python was configured to use ASCII as encoding for the environment. Consult https://click.palletsprojects.com/unicode-support/ for mitigation steps.

Yet when I take the command that I try to run in do shell script and paste it directly into terminal, the command works absolutely fine. Any ideas?

**Comment by samschott** [comment, marked as answer]
Yes, I've seen this before. We use [click](https://click.palletsprojects.com) for most of the command line interface and click requires a UTF-8 encoding of the environment. This is typically set by environment variables such as `LC_CTYPE=UTF-8` in your command line by default on macOS.

It looks like the environment in which automator runs has a ASCII encoding configured. There probably is a way to change that but I'm not particularly familiar with automator or AppleScript.

**Comment by theboyler** [comment]
Thanks Sam - this is helpful, and I set off to find scrips that would convert output from an AppleScript action in automator. 

In the course of testing those, something seems to have happened to the CLI. I've uninstalled it, and reinstalled it, and when installed, whatever command is given seems to execute no actions, even in terminal. I just try basic stuff like maestral ls and I just get another % prompt...

**Reply by samschott** [reply]
Oh, that is strange. Is only Maestral affected or other CLI commands as well? Is it possible that output is somehow redirected?

Installing and uninstalling the CLI only creates (or removes) a symlink at `/usr/local/bin/maestral` to the actual CLI executable at `/Applications/Maestral.app/Contents/MacOS/maestral-cli`. This should therefore not affect anything whatsoever, you might as well use the CLI executable directly.

**Comment by prc1212** [comment]
I had the same problem, with AppleScript. My workaround is to run the shell scripts from a Keyboard Maestro macro, which can be embedded in my AppleScript script. Not so elegant but it works perfectly well!
